server/src/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3378,6 +3379,10 @@ private String signRequest(final String request, final String key) {
         final Integer apiLimitInterval = Integer.valueOf(_configDao.getValue(Config.ApiLimitInterval.key()));
         final Integer apiLimitMax = Integer.valueOf(_configDao.getValue(Config.ApiLimitMax.key()));
 
+        // Check if users can view destroyed vm's, expunge and/or recover them. Admins can always do so.
```

Comment:
not really a comment that enlightens us beyond the naming of the variables, so could be omitted.